thingworx-import/Entities.xml

```diff
@@ -0,0 +1,885 @@
+        }
+
+        // Get tag data type
+        var tagType;
+        if ("type" in tag) {
+            if (tag.type === 0) {
+                // Ewon Boolean tag converts to Thingworx Boolean
+                tagType = "BOOLEAN";
+            } else if (tag.type === 1) {
+                // Ewon Float tag converts to Thingworx Number
+                tagType = "NUMBER";
+            } else if (tag.type === 2) {
+                // Ewon Integer tag converts to Thingworx Integer
+                tagType = "INTEGER";
+            } else if (tag.type === 3) {
+                // Ewon DWOARD tag converts to Thingworx Number
+                tagType = "NUMBER";
+            } else if (tag.type === 4) {
+                // Ewon String tag converts to Thingworx String
+                tagType = "STRING";
+            } else {
+                // Unknown tag type, default to Thingworx Number
+                logger.warn("Unknown tag type (" + tag.type + ") for tag: " + tag.name);
+                tagType = "NUMBER";
+            }
+        } else {
+            logger.warn("The connected Ewon device is not sending tag type information. Check that the Ewon Thingworx Connector has been updated on the Ewon device!");
+        }
+
+        // Add datapoint to Thing
+        me.InsertDataPoint({
+            tagName: tag.name /* STRING */ ,
+            ewonName: ewonName /* STRING */ ,
+            tagValue: tag.value /* STRING */ ,
+            timestamp: tag.timestamp /* STRING */ ,
+            tagQuality: tagQuality /* STRING */ ,
+            tagType: tagType /* STRING */ ,
+            isTalk2MDevice: false /* BOOLEAN */
+        });
+    }
+}
+run();
+
+// Registers that an update has been received
+me.ProcessTimeSinceUpdate({
+    hasUpdated: true
+});]]></code></Row></Rows></ConfigurationTable></ConfigurationTables></ServiceImplementation><ServiceImplementation description="" handlerName="Script" name="Talk2MSyncData"><ConfigurationTables><ConfigurationTable dataShapeName="" description="" isMultiRow="false" name="Script" ordinal="0"><DataShape><FieldDefinitions><FieldDefinition baseType="STRING" description="code" name="code" ordinal="0"/></FieldDefinitions></DataShape><Rows><Row><code><![CDATA[/*
+ * Talk2MSyncData gets and inserts all logged data from DataMailbox.
+ *
+ * @author HMS Networks, MU Americas Solution Center
+ */
+function parseEwon(ewon) {
+    // Gets Ewon name without leading/trailing spaces
+    var ewonName = ewon.name.trim();
+
+    // Loop through each tag listed
+    ewon.tags.forEach((tag) => {
+        var i;
+        if (tag.history) {
+            // Process each historical data point
+            tag.history.forEach((datapoint) => {
+
+                // Get tag data quality
+                var tagQuality = "GOOD";
+                if (datapoint.quality === "bad") {
+                    tagQuality = "BAD";
+                } else if (datapoint.quality === "unknown") {
+                    tagQuality = "UNKNOWN";
```

Comment:
Cant you just do a .toUpperCase() here? And why is the default case good and not unknown? If you get a datapoint.quality you dont expect i would assume that would be unknown rather than good.

Reply:
Talk2M doesnt return "quality" if it is good. That is only present if its not good.